app.rb

```diff
@@ -0,0 +1,124 @@
+require './book'
+require './student'
+require './teacher'
+require './rental'
+
+class App
+  def initialize
+    @books = []
+    @people = []
+  end
+
+  def add_book
+    puts
+    print 'Title: '
+    title = gets.chomp.strip.capitalize
+    print 'Author: '
+    author = gets.chomp.strip.capitalize
+    @books << Book.new(title, author)
+  end
+
+  def list_books
+    puts
+    @books.each_with_index do |book, index|
+      puts "#{index}) Title: \"#{book.title}\", Author: #{book.author}"
+    end
+  end
+
+  def create_person
+    puts
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the number]: '
+    choice = gets.chomp.strip.to_i
+    case choice
+    when 1
+      create_student
+    when 2
+      create_teacher
+    end
+  end
+
+  def create_teacher
+    puts
+    print 'Age: '
+    age = gets.chomp.strip.to_i
+    while age <= 0 || age >= 100
+      print 'Please input a valid age (1 - 100): '
+      age = gets.chomp.strip.to_i
+    end
+    print 'Name: '
+    name = gets.chomp.strip.capitalize
+    print 'Specialization: '
+    specialization = gets.chomp.strip
+    @people << Teacher.new(age, specialization, name: name)
+    puts 'Teacher created successfully'
+  end
+
+  def create_student
+    puts
+    print 'Age: '
+    age = gets.chomp.strip.to_i
+    while age <= 0 || age >= 100
+      print 'Please input a valid age: (1 - 100): '
+      age = gets.chomp.strip.to_i
+    end
+    print 'Name: '
+    name = gets.chomp.strip.capitalize
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp.strip.upcase
+    case permission
+    when 'Y'
+      permission = true
+    when 'N'
+      permission = false
+    end
+    @people << Student.new(age, nil, parent_permission: permission, name: name)
+    puts 'Student created successfully'
+  end
+
+  def create_rental
+    puts
+    puts 'Select a book from the following list by number'
+    list_books
+    book_choice = gets.chomp.to_i
+    while book_choice.negative? || book_choice >= @books.length
+      print "Please enter a number within 0 - #{@books.length - 1} range: "
+      book_choice = gets.chomp.to_i
+    end
+    book = @books[book_choice]
+    puts 'Select a person from the following list by number (not id)'
+    list_people
+    person_choice = gets.chomp.to_i
+    while person_choice.negative? || person_choice >= @people.length
+      print "Please enter a number within 0 - #{@people.length - 1} range: "
+      person_choice = gets.chomp.to_i
+    end
+    person = @people[person_choice]
+    print 'Enter date of booking: (yyyy/mm/dd) : '
+    date = gets.chomp.strip
+    person.add_rental(date, book)
+  end
```

Comment:
-  Hi @NayanKaran I currently cannot create a rental, when I type in a number this is what I see ⬇️ 
![Screenshot_414](https://user-images.githubusercontent.com/69638013/187050114-dc2aefdb-1970-4ad0-a81c-cb3f8f90179e.png)

- [ ] Kindly make sure a user can create a rental, when I add a number the prompt is repeated it is not successful please fix this 👍🏿

Reply:
@prolajumokeoni Thank you for your suggestion. I have added the required corner case, https://github.com/NayanKaran/school-library/pull/5/commits/38c8b50c2738d1c5a29a87254fc84fd5a5b3a053